src/overlays/actors/ovl_En_Holl/z_en_holl.h

```diff
@@ -4,6 +4,35 @@
 #include "ultra64.h"
 #include "global.h"
 
+#define ENHOLL_GET_TYPE(thisx) (((thisx)->params >> 6) & 7)
+#define ENHOLL_GET_SWITCH_FLAG(thisx) ((thisx)->params & 0x3F)
+
+/**
+ * Two kinds of holls:
```

Comment:
A personal thing, but I'm still stuck on the mentality of thinking of holls as being 2d plane/portal you cross with extra things being attached to make the game work nice, instead of this newer idea of en_holls being a three dimensional "hallway" that enable travel either vertically or horizontally. Because of it every single instance of horizontal and vertical is backwards to me because e.g. `EnHoll_HorizontalVisibleNarrow` computations deal with vertical planes. 

So I would like it if this line was more direct in explaining the dichotomy of en_holl instances being split by direction of travel, something like this but... better

```suggestion
 * There are two kinds of holls, which describe the direction in which the player can travel between rooms:
```

And then I also would like it if the first line of each function that has horizontal/vertical in the name would just spell out that you're referring to the direction of travel in some way, just to hammer the concept in.

Reply:
The direction of the player movement is shown below in this comment, I think it would just duplicate it to also mention it here

And idk just forget about holls being 2d, I know it doesn't help there is one or two types that draw a 2d plane but it's also only one or two types, not all of them

Reply:
I didn't mean to suggest duplicating the information in this file, I was suggesting to hoist it up closer into the first line since it's important and it would make the text read better.

And I can't just "forget" about it because of how I visualize the computations, and like you said, `EnHoll_HorizontalVisibleNarrow` is actually visually represented as a vertical plane. If there's going to be a disconnect between visual representation and implementation names I think it's fair to repeat the information by the functions at least.

Reply:
What's the takeaway?

I think I wrote this block comment in a way that I think can be parsed quickly, "tree-like", since typically it should (?) be a quick glance to look at
Adding details at the top instead of at the leaves would go against that
Also style-wise if the top line for v/h was amended I'd like if the one for visual effects was similarly amended

Reply:
I dont think it needs any more special treatment in the comment. if you just read the next few lines youll see the information you are looking for.

If there is a way we can fit that information in a nice way near the functions as well, we can possibly come to a compromise to move this forward